app/Libraries/Database.php

```diff
@@ -72,84 +77,212 @@ public static function getInstance(array $credentials = null): object
   }
 
   /**
-   * Prepare statement with query
+   * Retrieves all database names from server
    *
-   * @param string $sql
-   * @return void
+   * @return array
    */
-  public function query(string $sql): void
+  public function allDatabaseNames(): array
   {
-    $this->stmt = $this->dbh->prepare($sql);
+    $stmt = self::$pdo->query('SHOW DATABASES;');
+    return $stmt->fetchAll(PDO::FETCH_OBJ);
   }
 
   /**
-   * Bind values
+   * Retrieves all table names for specified database
+   *
+   * @return array
+   */
+  public function getTablesFromServer($databaseName): array
+  {
+    try {
+      $stmt = self::$pdo->query("SHOW TABLES FROM $databaseName;");

```

Comment:
Same as above: $stmt -> $statement
Also, you are missing bindings here. `:databaseName` instead of `$databaseName` for security reasons.